### PR TITLE
Add address helpers to protocol contracts

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Add address json
- Add address helpers
- Move to networks pkg

### CODE REVIEW

File: lib/address.tools.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  | "goerli_testnet"
+  | "mumbai_testnet"
+  | "zeta_testnet";
+
+export declare type ZetaProtocolMainNetwork = "etherum_mainnet";
+export declare type ZetaProtocolNetwork = ZetaProtocolMainNetwork | ZetaProtocolTestNetwork;
+
+export declare type ZetaProtocolEnviroment = "mainnet" | "testnet";
+
+export declare const isProtocolNetworkName: (str: string) => str is ZetaProtocolNetwork;
```
Comment (Mfigueira) [inline lib/address.tools.ts:19]:
```suggestion
export type ZetaProtocolTestNetwork =
  | "baobab_testnet"
  | "bsc_testnet"
  | "goerli_testnet"
  | "mumbai_testnet"
  | "zeta_testnet";

export type ZetaProtocolMainNetwork = "etherum_mainnet";

export type ZetaProtocolNetwork = ZetaProtocolMainNetwork | ZetaProtocolTestNetwork;

export const zetaProtocolNetworks: ZetaProtocolNetwork[] = [
  "baobab_testnet",
  "bsc_testnet",
  "etherum_mainnet",
  "goerli_testnet",
  "mumbai_testnet",
  "zeta_testnet",
];

export const isProtocolNetworkName = (str: string): str is ZetaProtocolNetwork =>
  zetaProtocolNetworks.includes(str as ZetaProtocolNetwork);
```

**Comment by fadeev** [comment]
What is the equivalent of the zeta token consumer in the new addresses file?

https://github.com/zeta-chain/example-contracts/blob/54645ad56400e5ad6bc739e80b9a939057feef30/messaging/message/tasks/deploy.ts#L52-L61

**Comment by fadeev** [comment]
Can we, please, add system contract address to `zeta_testnet`?

https://github.com/zeta-chain/example-contracts/blob/54645ad56400e5ad6bc739e80b9a939057feef30/omnichain/swap/tasks/deploy.ts#L5

File: data/addresses.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "tss": "",
+    "tssUpdater": "",
+    "zetaToken": "",
+    "immutableCreate2Factory": "0x095a03c6a68137fE9a566bBc3e552F299d8b886d"
+  },
+  "zeta_testnet": {
+    "connector": "0x239e96c8f17C85c30100AC26F635Ea15f23E9c67",
+    "tss": "0x7c125C1d515b8945841b3d5144a060115C58725F",
+    "tssUpdater": "0x7274d1d5dddef36aac53dd45b93487ce01ef0a55",
+    "zetaToken": "0x5F0b1a82749cb4E2278EC87F8BF6B618dC71a8bf"
```
Comment (fadeev) [inline data/addresses.json:37]:
```suggestion
    "zetaToken": "0x5F0b1a82749cb4E2278EC87F8BF6B618dC71a8bf",
    "systemContract": "0x91d18e54DAf4F677cB28167158d6dd21F6aB3921"
```

**Comment by fadeev** [comment]
Shall we also add ZRC20 addresses to connected chains?

**Comment by andresaiello** [comment]
> What is the equivalent of the zeta token consumer in the new addresses file?

I didn't deploy zeta consumer yet because we don't have liquidity pools yet

**Comment by andresaiello** [comment]
> Shall we also add ZRC20 addresses to connected chains?
@fadeev 
in fact maybe we should add another json and another methods with the system address and zrc20. to keep the same fields for every chain on this file and another for zevm

**Review by fadeev — APPROVED** [review]

**Pull request merged**